lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_l10n_helper.rb

```diff
@@ -0,0 +1,117 @@
+        #         - `:text` for the ASCII-plist file format (containing typical `"key" = "value";` lines)
+        #         - `:xml` for XML plist file format (can be used if machine-generated, especially since there's no official way/tool to generate the ASCII-plist file format as output)
+        #         - `:binary` for binary plist file format (usually only true for `.strings` files converted by Xcode at compile time and included in the final `.app`/`.ipa`)
+        #         - `nil` if the file does not exist or is neither of those format (e.g. not a `.strings` file at all)
+        #
+        def self.strings_file_type(path:)
+          format_desc, status = Open3.capture2('file', path)
+          return nil unless status.success?
+
+          case format_desc
+          when /Apple binary property list/ then return :binary
+          when /XML/ then return :xml
+          when /text/ then return :text
+          end
+        end
+
+        # Merge the content of multiple `.strings` files into a new `.strings` text file.
+        #
+        # @param [Array<String>] paths The paths of the `.strings` files to merge together
+        # @param [String] into The path to the merged `.strings` file to generate as a result.
+        # @return [Array<String>] List of duplicate keys found while validating the merge.
+        #
+        # @note For now, this method only supports merging `.strings` file in `:text` format
+        #       and basically concatenates the files (+ checking for duplicates in the process)
+        # @note The method is able to handle input files which are using different encodings,
+        #       guessing the encoding of each input file using the BOM (and defaulting to UTF8).
+        #       The generated file will always be in utf-8, by convention.
+        #
+        # @raise [RuntimeError] If one of the paths provided is not in text format (but XML or binary instead), or if any of the files are missing.
+        #
+        def self.merge_strings(paths:, output_path: nil)
+          duplicates = []
+          Tempfile.create('wpmrt-l10n-merge-', encoding: 'utf-8') do |tmp_file|
+            all_keys_found = []
+
+            tmp_file.write("/* Generated File. Do not edit. */\n\n")
+            paths.each do |input_file|
+              fmt = strings_file_type(path: input_file)
+              raise "The file `#{input_file}` does not exist or is of unknown format." if fmt.nil?
+              raise "The file `#{input_file}` is in #{fmt} format but we currently only support merging `.strings` files in text format." unless fmt == :text
+
+              string_keys = read_strings_file_as_hash(path: input_file).keys
+              duplicates += (string_keys & all_keys_found) # Find duplicates using Array intersection, and add those to duplicates list
+              all_keys_found += string_keys
+
+              tmp_file.write("// MARK: - #{File.basename(input_file)}\n\n")
```

Comment:
Nitpick: The header comment pasted above uses `/* */`. Is it possible to be consistent with the comment format? I assume Xcode would be able to parse these `MARK: -` even if in a `/* */`-kind comment.

Reply:
Never tried using `MARK` within `/* */`, will double-check 👍

Reply:
Seems it works just as well, so fixed this in 0354867 🙂